site/_posts/2021-03-16-securely-implement-oauth-vuejs.md

```diff
@@ -1174,32 +1221,310 @@ methods: {
+
+```vue
+data() {
+  return {
+    email: null,
+    body: null,
+    boolShowSignOut: false,
+    authState: null
+  }
+},
+```
+
+In the mounted function we are going to add `authState` and `boolShowSignOut`:
+
+```vue
+mounted() {
+  fetch(`http://localhost:9000/user`, {
+    credentials: "include" // fetch won't send cookies unless you set credentials
+  })
+  .then((response) => response.json())
+  .then((data) => {
+    if (data.authState == "Authorized"){
+      this.email = data.introspectResponse.email;
+      this.body = data.body;
+      this.boolShowSignOut = true
+    }
+    else if (data.authState == "notAuthorized"){
+      this.boolShowSignOut = true
+    }
+    else if (data.authState == "notAuthenticated"){
+      this.boolShowSignOut = false
+    }
+    this.authState = data.authState
+  });
+}
+```
+
+#### Add to your `Greeting.vue`
+
+Additionally, we will add `authState` to `Greeting.vue` (as well as some small styling):
+
+```vue
+<template>
+  <div className="greet">
+    <h3 v-if="email">Welcome{% raw %} {{email}} {% endraw %}</h3>
+    <h3 v-else>You are not logged in</h3>
+    <div>
+      <u>authState:</u>
+    </div>
+    <div class="authStateBox">{% raw %} {{authState}} {% endraw %}</div>
+  </div>
+</template>
+<script>
+export default {
+  name: 'Greet',
+  props: ["email", "authState"],
+};
+</script>
+<style>
+* {
+  margin-top: 30px;
+  text-align: center;
+  font-size: 20px;
+  font-family: 'Courier New', Courier, monospace;
+}
+
+.authStateBox {
+  margin: 20px;
+  background-color: lightcoral;
+  border-radius: 25px;
+}
+</style>
+```
+
+#### Add to your `Login.vue`
+In our `Login.vue` we will update to conditionally render on the `boolShowSignOut`: 
+
+```vue
+<template>
+  <h1 v-if="boolShowSignOut"><a href='http://localhost:9000/logout'>Sign Out</a></h1>
+  <h1 v-else><a href='http://localhost:9000/login'>Sign In</a></h1>
+</template>
+<script>
+export default {
+  name: "Login",
+  props: ["boolShowSignOut"],
+};
+</script>
+```
+
+With this code (and a small amount of styling code for a logo) our application should look as follows:
+
+#### Authorized
+
+{% include _image.liquid src="/assets/img/blogs/oauth-vuejs/user-authorized.png" alt="User authorized and authenticated." class="img-fluid" figure=false %}
+
+#### notAuthorized
+
+{% include _image.liquid src="/assets/img/blogs/oauth-vuejs/user-not-authorized.png" alt="User not authorized." class="img-fluid" figure=false %}
+
+#### notAuthenticated
+
+{% include _image.liquid src="/assets/img/blogs/oauth-vuejs/user-not-authen.png" alt="User not authenticated." class="img-fluid" figure=false %}
+
 ## Conclusion
 
 Congrats, you have built a Vue application which allows a user to log in, log out, and modify their user data. This article is a foundation for implementing OAuth using FusionAuth. There are a bunch of other features, components, and routes that you could add to expand this application.
```

Comment:
You might want to add a note about your last section that you added (displaying different messages if a user isn't registered for an application).

Reply:
Also, I'd add another 'do next idea' which is 'store the access token as a cookie rather than in a session'.